build-system/tasks/report-test-results.js

```diff
@@ -20,11 +20,78 @@
 
 'use strict';
 
+const fetch = require('node-fetch');
+const fs = require('fs').promises;
 const log = require('fancy-log');
-const {green} = require('ansi-colors');
+const path = require('path');
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../common/travis');
+const {cyan, green, yellow} = require('ansi-colors');
+
+const REPORTING_API_URL = 'https://amp-test-cases.appspot.com/report';
+
+async function getReport(testType) {
+  const karmaReportString = await fs.readFile(
+    `result-reports/${testType}.json`
+  );
+  const karmaReportJson = JSON.parse(karmaReportString);
```

Comment:
nit: this is one of very few situations where using `then` in conjunction with `await` could actually work nicely, ie.

```javascript
async function getReport(testType) {
  const report = await fs.readFile(`result-reports/${testType}.json`)
      .then(JSON.parse);
  return addJobAndBuildInfo(testType, report);
}
```

Reply:
Addendum: `require` also natively supports JSON and caches across invocations, so I wouldn't be opposed to the following:

```javascript
async function getReport(testType) {
  return addJobAndBuildInfo(
    testType,
    require(`result-reports/${testType}.json`)
  );
}
```

Reply:
Oh, wow, I didn't know require worked for that. I think I prefer the first version, but if you feel strongly about it I'll do the second one.

Reply:
Done implementing the first version